File: doc/codes/csharp/examples/api/compute/pattern/bessel-example.cs
using System.Numerics;
using AUTD3;
using static AUTD3.Units;

// HIDE
namespace DocSamples.ApiComputePatternBesselExample;

internal static class Sample
{
    internal static void Run()
    {
        // HIDE_END
var geometry = new Geometry(new[] { new Autd3(Vector3.Zero) });

var phases = geometry.PhaseBuffer();

Pattern.Bessel(
    geometry,
    geometry.Center + new Vector3(0.0f, 0.0f, 150.0f),
    Vector3.UnitZ,
    18.0f * deg,
    Pattern.Wavelength(340.0f * m / s),
    phases
);
        // HIDE
    }
}
// HIDE_END

File: doc/codes/python/examples/api/compute/pattern/bessel-example.py
import numpy as np
from autd3.geometry import Autd3, Geometry
from autd3.units import deg, m, s
from autd3_pattern import bessel, wavelength

geometry = Geometry([Autd3([0.0, 0.0, 0.0], [1.0, 0.0, 0.0, 0.0])])

phases = geometry.phase_buffer()

bessel(
    geometry,
    geometry.center() + np.array([0.0, 0.0, 150.0]),
    np.array([0.0, 0.0, 1.0]),
    18.0 * deg,
    wavelength(340 * m / s),
    phases,
)
